cloud/cloudstack/cs_loadbalancer_rule.py

```diff
@@ -0,0 +1,290 @@
+        args = {}
+        args['ipaddress'] = ip_address
+        args['account'] = self.get_account(key='name')
+        args['domainid'] = self.get_domain(key='id')
+        args['projectid'] = self.get_project(key='id')
+        ip_addresses = self.cs.listPublicIpAddresses(**args)
+
+        if ip_addresses:
+            self.ip_address = ip_addresses['publicipaddress'][0]
+        return self._get_by_key(key, self.ip_address)
+
+    def get_rule(self, **kwargs):
+        rules = self.cs.listLoadBalancerRules(**kwargs)
+        if rules:
+            return rules['loadbalancerrule'][0]
+
+    def _get_common_args(self):
+        return {
+            'account': self.get_account(key='name'),
+            'domainid': self.get_domain(key='id'),
+            'projectid': self.get_project(key='id'),
+            'zoneid': self.get_zone(key='id'),
+            'publicipid': self.get_ip_address(self.module.params.get('public_ip'), key='id'),
+            'name': self.module.params.get('name'),
+        }
+
+    def create_lb_rule(self):
+        args = self._get_common_args()
+        rule = self.get_rule(**args)
+        algorithm = self.module.params.get('algorithm')
+        privateport = self.module.params.get('private_port')
+        publicport = self.module.params.get('public_port')
+        if not all([algorithm, privateport, privateport]):
+            self.module.fail_json(msg="algorithm, private_port and public_port are required to create a rule")
+        if not rule:
+            self.result['changed'] = True
+        if not rule and not self.module.check_mode:
+            args.update({
+                'algorithm': algorithm,
+                'privateport': privateport,
+                'publicport': publicport,
+                'cidrlist': self.module.params.get('cidr'),
+            })
+            res = self.cs.createLoadBalancerRule(**args)
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                res = self.poll_job(res, 'loadbalancer')
+            rule = res
+
+        return rule
+
+    def delete_lb_rule(self):
+        args = self._get_common_args()
+        rule = self.get_rule(**args)
+        if rule:
+            self.result['changed'] = True
+        if rule and not self.module.check_mode:
+            res = self.cs.deleteLoadBalancerRule(id=rule['id'])
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+            poll_async = self.module.params.get('poll_async')
+            if poll_async:
+                res = self._poll_job(res, 'loadbalancer')
+        return rule
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(required=True),
+            algorithm = dict(choices=['source', 'roundrobin', 'leastconn'], required=False),
```

Comment:
yes sure, I mean we should have a default no matter what it is, this saves some user input.
If there is no default, user have to define it anyway... so make a default, take whatever you like. :)